contracts/src/PDPService.sol

```diff
@@ -184,6 +193,9 @@ contract PDPService {
         if (needsChallengeEpoch) {
             nextChallengeEpoch[setId] = block.number + challengeFinality; 
         }
+
+        bytes memory extraData = abi.encode(rootData.length, firstAdded, rootData);
```

Comment:
I liked putting length first as it makes things easy to parse but realistically its probably overkill since any parsing of rootData will have the length.  What do you think @anorth ?

Reply:
We probably don't need the length.